src/actions/profile-view.js

```diff
@@ -123,6 +132,34 @@ export function changeCallTreeSearchString(searchString: string): Action {
   };
 }
 
+function _addAllDescendants(
+  tree: CallTree,
+  newSet: Set<IndexIntoCallNodeTable>,
+  callNodeIndex: IndexIntoCallNodeTable
+) {
+  tree.getChildren(callNodeIndex).forEach(childId => {
+    newSet.add(childId);
+    _addAllDescendants(tree, newSet, childId);
+  });
+}
+
+export function expandAllCallNodeDescendants(
+  threadIndex: ThreadIndex,
+  callNodeIndex: IndexIntoCallNodeTable,
+  tree: CallTree,
+  callNodeInfo: CallNodeInfo,
+  expandedCallNodeIndexes: Array<IndexIntoCallNodeTable>
+): Action {
+  const newSet = new Set(expandedCallNodeIndexes);
+  newSet.add(callNodeIndex);
+  _addAllDescendants(tree, newSet, callNodeIndex);
+
+  const expandedCallNodePaths = Array.from(newSet.values()).map(callNodeIndex =>
+    getCallNodePath(callNodeIndex, callNodeInfo.callNodeTable)
+  );
+  return changeExpandedCallNodes(threadIndex, expandedCallNodePaths);
```

Comment:
Don't know if it's idiomatic to return another action like this?

Reply:
I think it's fine